Config/CMakeIncludes/FindTasmanianCudaMathLibs.cmake

```diff
@@ -0,0 +1,65 @@
+# install cuBlas in /usr/lib while nvcc and the other libs sit in /usr/local/cuda
+    find_library(${Tasmanian_variable}  ${Tasmanian_library}
+                 HINTS "${Tasmanian_nvccroot}"
+                 HINTS "${Tasmanian_nvccroot}/lib/"
+                 HINTS "${Tasmanian_nvccroot}/${CMAKE_LIBRARY_ARCHITECTURE}/lib/"
+                 HINTS "${Tasmanian_nvccroot}/lib/${CMAKE_LIBRARY_ARCHITECTURE}"
+                 HINTS "${Tasmanian_nvccroot}/lib64/"
+                 HINTS "${Tasmanian_nvccroot}/${CMAKE_LIBRARY_ARCHITECTURE}/lib64/"
+                 HINTS "${Tasmanian_nvccroot}/lib64/${CMAKE_LIBRARY_ARCHITECTURE}")
+endmacro(Tasmanian_cuda_search_lib)
+
+get_filename_component(Tasmanian_nvccroot ${CMAKE_CUDA_COMPILER} DIRECTORY)
+get_filename_component(Tasmanian_nvccroot ${Tasmanian_nvccroot} DIRECTORY)
+
+# find shared and static versions of the math libraries
+Tasmanian_cuda_search_lib(Tasmanian_cublas_static   cublas_static)
+Tasmanian_cuda_search_lib(Tasmanian_cublas_device   cublas_device)
+Tasmanian_cuda_search_lib(Tasmanian_cublas_lt       cublasLt)
+Tasmanian_cuda_search_lib(Tasmanian_cublas_ltstatic cublasLt_static)
+Tasmanian_cuda_search_lib(Tasmanian_cublas          cublas)
+Tasmanian_cuda_search_lib(Tasmanian_cusparse_static cusparse_static)
+Tasmanian_cuda_search_lib(Tasmanian_cusparse        cusparse)
+Tasmanian_cuda_search_lib(Tasmanian_culibos         culibos)
+
+if ((NOT EXISTS ${Tasmanian_cublas}) OR (NOT EXISTS ${Tasmanian_cusparse}))
+    message(FATAL_ERROR "CMake seems to have found CUDA, but Tasmanian could not locate the CUDA math librariesc cuBlas and cuSparse!")
+endif()
+
+list(APPEND Tasmanian_cudamathlibs  ${Tasmanian_cusparse_static}
+                                    ${Tasmanian_cublas_static}
+                                    ${Tasmanian_cublas_device}
+                                    ${Tasmanian_culibos}
+                                    ${Tasmanian_cublas_lt}
+                                    ${Tasmanian_cublas_ltstatic}
+                                    ${Tasmanian_cusparse}
+                                    ${Tasmanian_cublas})
+
+# different versions of CUDA require different combination of cublas_device and cublas_lt libraries
+# in place of building a database with detailed library list for each version of cuda,
+# assume the correct libraries are present and ignore any missing ones
+list(FILTER Tasmanian_cudamathlibs EXCLUDE REGEX "-NOTFOUND$")
+
+# find the runtime libraries
+Tasmanian_cuda_search_lib(Tasmanian_cuda   cuda)
+Tasmanian_cuda_search_lib(Tasmanian_cudart cudart)
+
+list(APPEND Tasmanian_cudaruntime   ${Tasmanian_cuda}
+                                    ${Tasmanian_cudart})
+
+list(FILTER Tasmanian_cudaruntime EXCLUDE REGEX "-NOTFOUND$")
+
+message(STATUS "Tasmanian found CUDA math libraries: ${Tasmanian_cudamathlibs}")
+message(STATUS "Tasmanian found CUDA runtime libraries: ${Tasmanian_cudaruntime}")
```

Comment:
So we just gather all the ones that are available we don't do something like
```CMake
find_package_handle_standard_args(TasmanianCudaMathLibs DEFAULT_MSG
  Tasmanian_cusparse_static Tasmanian_cublas_static ...)
```

Reply:
Not all libraries are required, different versions of cuda use different combination of auxiliary static/Lt/device libraries. The main cublas and cusparse are required, the rest ... I assume what's needed is present, otherwise I have to build a database for each cuda version.

I used the `find_package_handle_standard_args` for cublas and cusparse, to handle the `Required` libs. But the macro only prints the first library, so I will keep the verbose message for now. May suppress it for the release, but it is good for debugging for now.